### PR TITLE
DRAFT Code style automation PLEASE DO NOT MERGE

### PR DESCRIPTION
**Pull request by iainfogg**
This closes issue #313 .

I suggest we use the issue for discussing what standards etc. that we want, then use this PR to review the actual implementation, to try and keep it cleaner.

### CODE REVIEW

**Comment by iainfogg** [comment]
@springfall2008 once you've approved this, I will do a PR on `main` to fix any issues which these checks find, so that you have clean branch BEFORE we merge this in to `main`. As a result, I'm leaving this as DRAFT so this doesn't accidentally get merged in until after that's been done.

Also, this relies on that pre-commit.com/lite thing being set up on your repo.

**Review by springfall2008 — APPROVED** [review]

**Comment by iainfogg** [comment]
I'm taking this one out of draft @springfall2008 , but please don't merge this one. I'm trying to test out the standards checking running on your repo.

**Comment by iainfogg** [comment]
@springfall2008 I'm merging this from the code-style branch in my repo to a branch of the same name in your repo, to test out the code style pipeline. This isn't putting it into main any more (I'll do that separately another time).

**Pull request merged**
